src/library_sdl.js

```diff
@@ -2394,6 +2394,19 @@ var LibrarySDL = {
       if (!SDL.audioContext) throw 'Web Audio API is not available!';
       SDL.audio.nextPlayTime = 0; // Time in seconds when the next audio block is due to start.
       
+      var sources = [];
```

Comment:
I've done a few changes: moved up the definition of the function + variable to just after `SDL.audio.caller`, and renamed the variable+function. I was a bit confused about your comment, regarding the use of the words global/local, and level, so let me know if you would like something different/more here.